pkg/host/root.go

```diff
@@ -187,14 +189,50 @@ func (h *Manager) patchHost(host *Host) {
 	//TODO
 }
 
-func (h *Manager) addHost(ctx context.Context, host *Host) {
-	clog := log.New()
-
+// createHostLogger creates a logger for a host that inherits from the base logger
+// while allowing host-specific overrides like log level
+func (h *Manager) createHostLogger(host *Host) *log.Entry {
+	// If the host specifies a custom log level, create a new logger instance
 	if host.LogLevel != nil {
-		clog.SetLevel(*host.LogLevel)
+		// Create a new logger with the host-specific level
+		// We need to create a new logger instance to avoid affecting the base logger
+		hostLogger := log.NewEntry(h.Logger.Logger)
+		hostLogger.Logger.SetLevel(*host.LogLevel)
+
+		// Copy the base logger's fields but exclude "component" since hosts should have their own context
+		for k, v := range h.Logger.Data {
+			if k != "component" {
+				hostLogger = hostLogger.WithField(k, v)
+			}
+		}
+
+		// Add the host field
+		return hostLogger.WithField("host", host.Host)
 	}
 
-	host.logger = clog.WithField("host", host.Host)
+	// For normal case, create a new logger entry without the "component" field
+	hostLogger := log.NewEntry(h.Logger.Logger)
+
+	// Copy the base logger's fields but exclude "component"
+	for k, v := range h.Logger.Data {
+		if k != "component" {
+			hostLogger = hostLogger.WithField(k, v)
+		}
+	}
+
+	return hostLogger.WithField("host", host.Host)
+}
+
+func (h *Manager) addHost(ctx context.Context, host *Host) {
+	// Create a logger for this host that inherits base logger values
+	host.logger = h.createHostLogger(host)
+
+	// Add additional useful fields for host context
+	host.logger = host.logger.WithFields(log.Fields{
+		"host_pattern": host.Host,
```

Comment:
technically `host_pattern` and `host` have the same value, so we can most likely drop this `host_pattern`